src/query/src/sql.rs

```diff
@@ -93,3 +106,124 @@ pub fn show_tables(stmt: ShowTables, catalog_manager: CatalogManagerRef) -> Resu
         .context(error::CreateRecordBatchSnafu)?;
     Ok(Output::RecordBatches(records))
 }
+
+pub fn describe_table(stmt: DescribeTable, catalog_manager: CatalogManagerRef) -> Result<Output> {
+    let schema = DEFAULT_SCHEMA_NAME;
+    let schema = catalog_manager
+        .schema(DEFAULT_CATALOG_NAME, schema)
+        .context(error::CatalogSnafu)?
+        .context(error::SchemaNotFoundSnafu { schema })?;
+    let table = schema
+        .table(&stmt.table_name)
+        .context(error::CatalogSnafu)?
+        .context(error::TableNotFoundSnafu {
+            table: &stmt.table_name,
+        })?;
+
+    let res_schema = Arc::new(Schema::new(vec![
+        ColumnSchema::new(
+            COLUMN_NAME_COLUMN,
+            ConcreteDataType::string_datatype(),
+            false,
+        ),
+        ColumnSchema::new(
+            COLUMN_TYPE_COLUMN,
+            ConcreteDataType::string_datatype(),
+            false,
+        ),
+        ColumnSchema::new(
+            COLUMN_NULLABLE_COLUMN,
+            ConcreteDataType::string_datatype(),
+            false,
+        ),
+        ColumnSchema::new(
+            COLUMN_DEFAULT_COLUMN,
+            ConcreteDataType::string_datatype(),
+            false,
+        ),
+        ColumnSchema::new(
+            COLUMN_SEMANTIC_TYPE_COLUMN,
+            ConcreteDataType::string_datatype(),
+            false,
+        ),
+    ]));
+
+    let table_info = table.table_info();
+    let columns_schemas = table_info.meta.schema.column_schemas();
+    let columns = vec![
+        describe_column_names(columns_schemas),
+        describe_column_types(columns_schemas),
+        describe_column_nullables(columns_schemas),
+        describe_column_defaults(columns_schemas),
+        describe_column_semantic_types(columns_schemas, &table_info.meta.primary_key_indices),
+    ];
+    let records = RecordBatches::try_from_columns(res_schema, columns)
+        .context(error::CreateRecordBatchSnafu)?;
+    Ok(Output::RecordBatches(records))
+}
+
+fn describe_column_names(columns_schemas: &[ColumnSchema]) -> VectorRef {
+    Arc::new(StringVector::from(
```

Comment:
use [`from_iter()`](https://github.com/GreptimeTeam/greptimedb/blob/develop/src/datatypes/src/scalars.rs#L97-L103) can reduce a copy here

Reply:
`from_iterator` requires `RefItem`, so I can't use `String`s here. Is there a way to overcome this?

Reply:
```rust
    Arc::new(StringVector::from_iterator(
        columns_schemas.iter().map(|cs| cs.name.as_str()),
    ))
```
`as_str()` looks work. It converts `&String` to `&str`

Reply:
Thx, it's working, but I had problem with `describe_column_defaults`. It creates a string in map fn, so it's impossible to map to reference. I've changed other functions though.